src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+
+  for (const action of actions) {
+    switch (action.type) {
+      case add:
+        for (const param in action.extraData) {
+          state[param] = action.extraData[param];
+        }
+        break;
+
+      case remove:
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      default:
```

Comment:
This is not default case, it's "clear" action type. Default should be for situation when action type is neither add, remove nor clear, and throw user error informing that provided action type is not supported